src/Pagination/FractalDoctrinePaginatorAdapter.php

```diff
@@ -94,28 +81,30 @@ public function getPerPage(): int
      */
     public function getUrl(int $page): string
     {
-        $url = $this->request->getUri();
         $params = $this->request->getQueryString($page);
-
-        return $url . '?' . $params;
+        return $this->url . '?' . $params;
     }
 
-    /**
-     * Get the paginator instance.
-     * @phpstan-return Paginator<Entity>
-     */
-    public function getPaginator(): Paginator
+    public function __serialize(): array 
     {
-        return $this->paginator;
+        return [
+            'total' => $this->total,
+            'url' => $this->url,
+            ...$this->request->getPageDetails()->getParamsArray(),
+            ...$this->request->getIncludes()->getParamsArray(),
+            ...$this->request->getSortDetails()->getParamsArray(),
+            'filters' => $this->request->getFiltersDetails()->toArray(),
+            'fields' => $this->request->getFields()->toArray(),
+            'excludes' => $this->request->getExcludes()->toArray(),
+        ];
     }
 
-    /**
-     * Get the the route generator.
-     *
-     * @return callable
-     */
-    public function getRouteGenerator()
+    public function __unserialize(array $data): void
     {
-        return $this->routeGenerator;
+        $this->total = $data['total'];
+        $this->url = $data['url'];
+        unset($data['total']);
+        unset($data['url']);
+        $this->request = RequestParams::fromArray($data);
```

Comment:
This has exposed a bug upstream in `giadc/json-api-request`.

If there is no sorting data, `RequestParams` uses an empty string here:

https://github.com/giadc/json-api-request/blob/master/src/Requests/RequestParams.php#L195

Exploding an empty string here…

https://github.com/giadc/json-api-request/blob/master/src/Requests/Sorting.php#L56

…creates the following for some mysterious PHP reason:

```
[
    0 => '',
]
```

That causes bugs further down the line. We can avoid this issue by checking the `strlen()` of the string in the `Sorting` constructor.

Reply:
completely forgot this pr existed. https://github.com/giadc/json-api-request/pull/8

Reply:
@rmarganti ^